config/v1/types_ingress.go

```diff
@@ -41,13 +43,122 @@ type IngressSpec struct {
 	// certificate.
 	// +optional
 	AppsDomain string `json:"appsDomain,omitempty"`
+
+	// componentRoutes is an optional list of routes configured by the cluster-admin to
+	// override the default hostname and serving certificates. The namespace and name of
+	// each componentRoute should match an existing componentRoute defined in the status.
+	//
+	// To determine the set of configurable Routes, look at namespace and name of entries in the
+	// .status.componentRoutes, where participating operators place current route status.
+	// +optional
+	ComponentRoutes []ComponentRouteSpec `json:"componentRoutes,omitempty"`
 }
 
+// consumingUser is an alias for string which we add validation to. Currently only service accounts are supported.
```

Comment:
If the pattern restricts values to service account names now, is it possible to relax the pattern to include other names in the future?

Reply:
Yes, we simply want to limit the scope of possible inputs for the initial implementation.